pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -301,6 +297,21 @@ func (c *InstallerController) manageInstallationPods(operatorSpec *operatorv1.St
 
 		// if we are in a transition, check to see whether our installer pod completed
 		if currNodeState.TargetRevision > currNodeState.CurrentRevision {
+
+			// get current revision status
+			_, err := c.configMapsGetter.ConfigMaps(c.targetNamespace).Get(fmt.Sprintf("revision-status-%d", currNodeState.TargetRevision), metav1.GetOptions{})
+			if apierrors.IsNotFound(err) && operatorStatus.LatestAvailableRevision > currNodeState.CurrentRevision {
+				// current revision status was deleted, set target to latest available revision
+				newCurrNodeState := currNodeState.DeepCopy()
+				newCurrNodeState.TargetRevision = operatorStatus.LatestAvailableRevision
+				if b, err, done := c.updateNodeStatus(newCurrNodeState, currNodeState, fmt.Sprintf("target revision %d no longer available", currNodeState.TargetRevision)); done {
+					return b, err
+				}
+			}
+			if err != nil {
+				return true, err
+			}
+
 			if err := c.ensureInstallerPod(currNodeState.NodeName, operatorSpec, currNodeState.TargetRevision); err != nil {
 				c.eventRecorder.Warningf("InstallerPodFailed", "Failed to create installer pod for revision %d on node %q: %v",
 					currNodeState.TargetRevision, currNodeState.NodeName, err)
```

Comment:
Why don't we just fall through here and let `newNodeStateForInstallInProgress` notice the missing installer, at least for the `pendingNewRevision==true` case? I.e. if setting the owner references fails, `ensureInstallerPod` will fail. If there is another latest revision, we mark the target version as failed, and continue with the latest version in the neyt sync. That would be a three liner change.

Reply:
Because when we get into the state we are testing for, the latest available revision is never updated. I'll take another look at the revision_controller and see if its simpler to fix this initial issue.

P.S. I also didn't want to hide that we were missing a revision by just deducing it as a side-effect of setting the owner ref.

Reply:
If we fall through we, we set the revision as failed. And because there is another one pending, we continue with that.

Reply:
> Because when we get into the state we are testing for, the latest available revision is never updated. I'll take another look at the revision_controller and see if its simpler to fix this initial issue.

The revision controller will create new revisions afaik because revisioned target configmaps and secrets are missing. It increases latest available version.

Was thinking about this:

```
if currNodeState.TargetRevision > currNodeState.CurrentRevision {
   pendingNewRevision := operatorStatus.LatestAvailableRevision > currNodeState.TargetRevision

   if err := c.ensureInstallerPod(currNodeState.NodeName, operatorSpec, currNodeState.TargetRevision); err != nil {
      c.eventRecorder.Warningf("InstallerPodFailed", "Failed to create installer pod for revision %d on node %q: %v", currNodeState.TargetRevision, currNodeState.NodeName, err)
      if !pendingNewRevision {
          return true, err
      }
      // fall through, newNodeStateForInstallInProgress will notice missing installer pod and update status
 }

newCurrNodeState, installerPodFailed, reason, err := c.newNodeStateForInstallInProgress(currNodeState, pendingNewRevision)
```

Simple, local change. What do I miss?